### ISSUE TITLE
Optimize cell table generation code

### ISSUE THREAD

**Issue by ngreenwald** [angelolab/ark-analysis#875]
**Is your feature request related to a problem? Please describe.**
The current code for generating a cell table is quite slow. Initially, since we only performed this action once, it wasn't a big deal. However, I have now written a couple helper functions that rely on this code, and the runtime is becoming more of an issue

**Describe the solution you'd like**
It's not clear to me which part of the code is the slowest part. My suspicion is that it's computing the per-cell [marker counts](https://github.com/angelolab/ark-analysis/blob/main/ark/segmentation/signal_extraction.py#L59), but I'm not sure. 

We should first perform some profiling to see where the problem is. If it's in that section of the code, it's possible that the function @JLrumberger wrote to compute the average model confidence per cell could be used here instead.

**Comment by alex-l-kong** [comment]
The main bottlenecks lie in the following two sections:

- https://github.com/angelolab/ark-analysis/blob/main/ark/segmentation/marker_quantification.py#L59-L61 (generating the single compartment region props data across all cells)
- https://github.com/angelolab/ark-analysis/blob/main/ark/segmentation/marker_quantification.py#L293-L325 (assigning the compartment features to each cell)

In both cases, no intermediate step takes a significantly long time (a few hundredths of a second at most for some regionprops functions, same for assigning compartment features). The bottleneck is inherent to the number of cells being run (ex. with 6000 cells, each requiring 0.02 seconds of processing, we're easily looking at a full 2 minutes per FOV).

This is a situation where I think multiprocessing should be considered. We can create cell tables in parallel since neither one depends on the other. On the MIBIAN, which has far more computational power than a laptop, memory shouldn't be as much of an issue.

**Comment by ngreenwald** [comment]
That makes sense, these are custom functions we wrote, so they're super slow. 

Can you investigate the following:
1. How much of a speedup we get if we remove the custom regionprops features, and just use the default ones
2. How much of a speedup we get if we completely remove all regionprops features, and just return the counts of each marker per cell?

**Comment by alex-l-kong** [comment]
1. Removing just the custom regionprops features saves about 30% of the time, these functions are fairly expensive especially when all computed at once.
2. Removing all the regionprops functions completely saves about 3 more seconds per FOV, the bottleneck of `assign_single_compartment_features` (which runs per cell, also runs `EXTRACTION_FUNCTION` to extract marker counts) still exists.

**Comment by ngreenwald** [comment]
And how long is a single FOV taking?

**Comment by alex-l-kong** [comment]
Depends on how many cells there are but on the example dataset our worst FOVs could take up to 3 minutes to complete. Those FOVs see a reduction down to about 1.25-1.5 minutes without the custom regionprops info.

**Comment by ngreenwald** [comment]
Got it. Will you talk to @JLrumberger so he can give you a pointer to the code he's using to extract model probabilities per cell? It may require GPU-paralellization, which would be a no-go for us, but I think it's worth checking to see if there's a CPU equivalent that's faster

**Comment by alex-l-kong** [comment]
@ngreenwald I talked with @JLrumberger and he's using Tensorflow for operations, he does PyTorch due to popularity and easier maintainability.

I tested it out on the Docker image without GPU acceleration (just plain `torch` commands) and unfortunately we're not seeing significant speedups compared to just `numpy`. This is probably something we'd have to up the ante on in terms of our internal Docker setup.

We should probably go the route you mentioned during last meeting, which is to allow the user to remove the option of extracting regionprops and marker intensity stats.
